G1ANT.Addon.UI/Api/XPathUIElementBuilder.cs

```diff
@@ -148,18 +144,14 @@ public object Operator(XPathOperator op, object left, object right)
                     {
                         return new CompareFunc((elem, index) =>
                         {
-                            string propValue = elem.FrameworkAutomationElement.ControlType.ToString().Replace("ControlType.", "");
-                            if (propValue != null)
-                                return propValue.Equals(right);
-                            return false;
+                            return elem.Properties.ControlType.IsSupported && elem.Properties.ControlType.Value.ToString().Replace("ControlType.", "").Equals(right);
                         });
                     }
                     if (UiAutomationElement.Id == en)
                     {
                         return new CompareFunc((elem, index) =>
                         {
-                            var propValue =  elem.FrameworkAutomationElement.ControlType.ValueOrDefault;
-                            return propValue.ToString().Equals(right); //Check if it's unknown?
+                            return elem.Properties.ControlType.IsSupported && elem.Properties.ControlType.ValueOrDefault.ToString().Equals(right);
```

Comment:
ControlType is Enum and always has a value so I think you can use ...ControlType.Value....